platform/v1alpha1/types.go

```diff
@@ -0,0 +1,76 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	TypeSourced = "Sourced"
+	TypeApplied = "Applied"
+
+	ReasonSourceFailed     = "SourceFailed"
+	ReasonSourceSuccessful = "SourceSuccessful"
+	ReasonApplyFailed      = "ApplyFailed"
+	ReasonApplySuccessful  = "ApplySuccessful"
+)
+
+// PlatformOperatorSpec defines the desired state of PlatformOperator.
+type PlatformOperatorSpec struct {
+	// PackageName contains the desired OLM-based Operator package name
+	// that is defined in an existing CatalogSource resource in the cluster.
+	//
+	// This configured package will be managed with the cluster's lifecycle.
```

Comment:
Would be good to add a comment about where a user can find the name of the package, and, if possible any restrictions. I know the package name is free form text, does it have any length restrictions? Can we add a max length here of say 1024 characters and be confident that wouldn't cause issues?

Reply:
I believe that the only restrictions are those set by Kube itself. As for explaining where a package's name can be a lot of this will be documentation around the feature for the initial pass. We're focusing on a few (two to three) packages upfront that will be served in the redhat-operators catalog. 

I'll update this comment to describe what I'm talking about here better.

Reply:
In general we try to avoid unbounded fields and add a length cap, do you think there's a reasonable upper limit we can impose here? 1024 chars? 2048?

Reply:
IIRC the default is a [DNS subdomain name](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names) which would mean a limit of 253 chars (which is also the longest option)

Reply:
Alright, I added a DNS subdomain pattern validation as well as a max limit of 56 (as imposed by the operator-sdk).